lte/gateway/c/session_manager/SessionStateEnforcer.cpp

```diff
@@ -136,15 +134,252 @@ void SessionStateEnforcer::handle_session_init_rule_updates(
+ * Go over session map and find the respective session of imsi
+ * Go over SessionState vector and dfnd the respective dnn (apn) 
+ * start terminating session process
+ */
+bool SessionStateEnforcer::m5g_release_session(
+      SessionMap& session_map, const std::string& imsi,
+      const std::string& dnn, SessionUpdate& session_update)
+{
+   bool exist = false;
+   auto map_element = session_map.find(imsi);
+   if((map_element == session_map.end())) {
+      MLOG(MERROR) << "No session found in SessionMap for IMSI " << imsi
+                   << " during releasing the session";
+      return false;
+   }
+
+   for (auto& session : map_element->second) {
+     auto config     = session->get_config();
+     auto session_id = session->get_session_id();
+     if (config.common_context.apn() == dnn){
+	/*Irrespective of any State of Session, release and terminate*/
+        SessionStateUpdateCriteria& uc = session_update[imsi][session_id];
+        MLOG(MINFO) 
+		<< "Start terminating the release request for IMSI "
+                << imsi << " having session id " << session_id
+		<< " current state of session is "
+                << session_fsm_state_to_str(session->get_state());
+	exist = true;
+        m5g_start_session_termination(imsi, session, dnn, uc);
+     } 
+   }
+   if(!exist) {
+     /* dnn  is not matching, nothing to proceed
+      * further to terminate session and return from here
+      */
+     MLOG(MERROR) << "Entry not found in the session context"
+                  << " IMSI " << imsi << " DNN " << dnn
+		  << " nothing to release. Return ";
+     return false;
+  }
+  return true;
+}
+
+/*Start processing to terminate respective session requested from AMF*/
+void SessionStateEnforcer::m5g_start_session_termination(
+      const std::string& imsi, const std::unique_ptr<SessionState>& session,
+      const std::string& dnn, SessionStateUpdateCriteria& uc)
+{
+   auto session_id = session->get_session_id();
+
+  /* update respective session's state and return from here before timeout
+   * to update session store with state and version
+   */
+  session->set_fsm_state(RELEASE, uc);
+  uint32_t cur_version = session->get_current_version();
+  session->set_current_version(cur_version++, uc);
+  MLOG(MINFO) << "During release state of session changed to "
+               << session_fsm_state_to_str(session->get_state())
+	       << " Current version is " << session->get_current_version();
+
+  /* Call for all rules to be de-associated from session 
+   * Message to UPF need to be filled from session i.e. of session's
+   * current_version_ 
+   * current_state_ and
+   * dnn can directly be used.
+   */
+  m5g_remove_rules_from_session_for_termination(imsi,
+		   session, dnn, uc);
+
+  /* Forcefully terminate session context on time out
+   * time out = 5000ms from sessiond.yml config file
+   */
+  MLOG(MINFO) << "Scheduling a force termination timeout for session_id "

```

Comment:
lets set this as `MDEBUG` as we do in 4g

Reply:
Currently we are using MINFO to get the messages during testing, else we need to change the log level in config file every time to test. We will do the needful of changing the log level once the basic code gets stabilized.